tests/test_remotedesktop.py

```diff
@@ -78,3 +78,154 @@ def test_remote_desktop_create_session_signal_closed(self):
+            session_handle=session.handle,
+            options=options,
+        )
+        assert response.response == 0
+
+        request = Request(self.dbus_con, rd_intf)
+        options = {}
+        response = request.call(
+            "Start",
+            session_handle=session.handle,
+            parent_window="",
+            options=options,
+        )
+        assert response.response == 0
+
+        with self.assertRaises(dbus.exceptions.DBusException) as cm:
+            fd = rd_intf.ConnectToEIS(
+                session.handle, dbus.Dictionary({}, signature="sv")
+            )
+        assert "Purposely failing ConnectToEIS" in cm.exception.get_dbus_message()
+
+    def test_remote_desktop_connect_to_eis_fail_notifies(self):
+        self.start_impl_portal()
+        self.start_xdp()
+
+        rd_intf = self.get_dbus_interface()
+        request = Request(self.dbus_con, rd_intf)
+        options = {
+            "session_handle_token": "session_token0",
+        }
+        response = request.call(
+            "CreateSession",
+            options=options,
+        )
+
+        assert response.response == 0
+
+        session = Session.from_response(self.dbus_con, response)
+        request = Request(self.dbus_con, rd_intf)
+        options = {
+            "types": dbus.UInt32(0x3),
+        }
+        response = request.call(
+            "SelectDevices",
+            session_handle=session.handle,
+            options=options,
+        )
+        assert response.response == 0
+
+        request = Request(self.dbus_con, rd_intf)
+        options = {}
+        response = request.call(
+            "Start",
+            session_handle=session.handle,
+            parent_window="",
+            options=options,
+        )
+        assert response.response == 0
+
+        for notifyfunc in [
+            {"name": "NotifyPointerMotion", "args": (1, 2)},
+            {"name": "NotifyPointerMotionAbsolute", "args": (0, 1, 2)},
+            {"name": "NotifyPointerButton", "args": (1, 1)},
+            {"name": "NotifyPointerAxis", "args": (1, 1)},
+            {"name": "NotifyPointerAxisDiscrete", "args": (1, 1)},
+            {"name": "NotifyKeyboardKeycode", "args": (1, 1)},
+            {"name": "NotifyKeyboardKeysym", "args": (1, 1)},
+            {"name": "NotifyTouchDown", "args": (0, 0, 1, 1)},
+            {"name": "NotifyTouchMotion", "args": (0, 0, 1, 1)},
+            {"name": "NotifyTouchUp", "args": (0,)},
+        ]:
+            with self.assertRaises(dbus.exceptions.DBusException) as cm:
+                func = getattr(rd_intf, notifyfunc["name"])
+                assert func is not None
+                func(
+                    session.handle,
+                    dbus.Dictionary({}, signature="sv"),
+                    *notifyfunc["args"]
+                )
+            # Not the best error message but...
+            assert (
+                "Session doesn't have access to a device of type"
```

Comment:
Maybe the error should be tweaked to be less specific, e.g. just  "Not allowed"?

Reply:
Follow-up commit changes this to `Session is not allowed to call NotifyPointer methods` and the same for `Keyboard` and `Touch`